MovieApp/MovieApp/App/Features/Movies/Repositories/MoviesRepository.swift

```diff
@@ -6,8 +6,22 @@ class MoviesRepository: MoviesRepositoryProtocol {
         self.networkDataSource = networkDataSource
     }
     
-    func fetchMovies(completion: @escaping (NetworkResult<MoviesNetworkModel, NetworkError>) -> Void) {
-        networkDataSource.fetchMovies(completion: completion)
+    func fetchMovies(completion: @escaping (NetworkResult<[MovieRepositoryModel], NetworkError>) -> Void) {
+        networkDataSource.fetchMovies { (result: NetworkResult<[MovieDataSourceModel], NetworkError>) in
+            switch result {
+            case .failure(let error):
+                print(error.localizedDescription)
+            case .success(let value):
+                let repositoryModel: [MovieRepositoryModel] = value.map { model -> MovieRepositoryModel in
+                    let imageUrl = NetworkConstants.imagePath + model.imageUrl
```

Comment:
Like we talked yesterday. Repo is only for deciding which data source to take or combining different data sources. Use Case is a business logic layer and you should put that inside UseCase

Reply:
I have it in use case and forgot to delete it from repo layer by mistake.